drivers: sensor: Migrate to new logger

punitvara opened the pull request:
Move all sensor drivers and samples to new logging system.

Signed-off-by: Punit Vara <punit.vara@intel.com>

nashif commented on drivers/sensor/adt7420/adt7420.h:
```diff
@@ -10,6 +10,7 @@
 #include <zephyr/types.h>
 #include <device.h>
 #include <gpio.h>
+#include <logging/log.h>
```
include this where it is being used, ie. in the .c file where logging functions and macros are being called.

punitvara commented:
recheck

punitvara committed:
drivers: sensor: Migrate to new logger

Move all sensor drivers and samples to new logging system.

Signed-off-by: Punit Vara <punit.vara@intel.com>

nashif commented on drivers/sensor/adt7420/adt7420.c:
```diff
@@ -11,8 +11,12 @@
 #include <kernel.h>
 #include <sensor.h>
 #include <misc/__assert.h>
+#include <logging/log.h>
```
move that below and after LOG_LEVEL is defined, otherwise you will get the default log level and not the one you are specifying below.

nordic-krch replied:
not needed any more. Now log level is resolved only when used so order does not matter.

nashif commented on tests/drivers/build_all/sensors_i_z.conf:
```diff
@@ -3,8 +3,8 @@ CONFIG_GPIO=y
-CONFIG_SYS_LOG=y
-CONFIG_SYS_LOG_SENSOR_LEVEL=4
+CONFIG_LOG=y
+CONFIG_SENSOR_LOG_LEVEL=4
```
should be CONFIG_SENSOR_LOG_LEVEL_DBG=y

nashif commented on samples/sensor/adxl372/prj.conf:
```diff
@@ -1,8 +1,8 @@
 CONFIG_ADXL372=y
 CONFIG_ADXL372_SPI=y
-CONFIG_SYS_LOG_SENSOR_LEVEL=3
+CONFIG_SENSOR_LOG_LEVEL=3
```
should be CONFIG_SENSOR_LOG_LEVEL_WRN=y

nashif commented on samples/sensor/amg88xx/prj.conf:
```diff
@@ -1,6 +1,6 @@
 CONFIG_STDOUT_CONSOLE=y
-CONFIG_SYS_LOG_SENSOR_LEVEL=1
-CONFIG_SYS_LOG=y
+CONFIG_SENSOR_LOG_LEVEL=1
```
should be CONFIG_SENSOR_LOG_LEVEL_INF=y,  also in other files below...

punitvara replied:
@nashif  All comments are addressed.

nordic-krch commented on drivers/sensor/lis2dh/lis2dh.c:
```diff
@@ -9,6 +9,41 @@
+#define LOG_LEVEL CONFIG_SENSOR_LOG_LEVEL
+LOG_MODULE_REGISTER(lis2dh);
+
+static int lis2dh_bus_configure(struct device *dev)
```
note that there is #10215 which will allow logging in headers. Is this function intended to be internal only? Previously was available through lis2dh.h.

punitvara replied:
@nordic-doki  This was used internal to driver so I moved to .c and I didn't know about #10215.  There is no use of this function externally as well. If you want me to place under .h I don't mind. 
Addressed all comments from you. Please go through PR once

nashif replied:
did you push? The PR still has the same issues

nordic-krch reviewed (dismissed):
There is one general comment. When LOG_MODULE_DECLARE is used, you need to define LOG_LEVEL as well. Otherwise default level is compiled in within that file.

nordic-krch commented on drivers/sensor/lsm6dsl/lsm6dsl_i2c.c:
```diff
@@ -9,10 +9,14 @@
 
 static u16_t lsm6dsl_i2c_slave_addr = CONFIG_LSM6DSL_I2C_ADDR;
 
+LOG_MODULE_DECLARE(LSM6DSL);
```
missing LOG_LEVEL

nordic-krch commented on drivers/sensor/lsm6dsl/lsm6dsl_shub.c:
```diff
@@ -10,9 +10,11 @@
 
 #include "lsm6dsl.h"
 
+LOG_MODULE_DECLARE(LSM6DSL);
```
missing `#define LOG_LEVEL`

nordic-krch commented on drivers/sensor/vl53l0x/vl53l0x_platform.c:
```diff
@@ -15,11 +15,9 @@
-#define SYS_LOG_LEVEL CONFIG_SYS_LOG_SENSOR_LEVEL
-#include <logging/sys_log.h>
-
+LOG_MODULE_DECLARE(VL53L0X);
```
missing `LOG_LEVEL`

nordic-krch commented on drivers/sensor/lsm6dsl/lsm6dsl_spi.c:
```diff
@@ -10,9 +10,11 @@
 
 #define LSM6DSL_SPI_READ		(1 << 7)
 
+LOG_MODULE_DECLARE(LSM6DSL);
```
missing `LOG_LEVEL`

nashif commented on samples/sensor/apds9960/prj.conf:
```diff
@@ -4,8 +4,8 @@ CONFIG_I2C=y
-CONFIG_SYS_LOG=y
-CONFIG_SYS_LOG_SENSOR_LEVEL=1
+CONFIG_LOG=y
+CONFIG_SENSOR_LOG_LEVEL=1
```
CONFIG_SENSOR_LOG_LEVEL_INF=y

nashif commented on samples/sensor/bmg160/prj.conf:
```diff
@@ -1,7 +1,7 @@
-CONFIG_SYS_LOG=y
-CONFIG_SYS_LOG_SENSOR_LEVEL=4
+CONFIG_LOG=y
+CONFIG_SENSOR_LOG_LEVEL=4
```
CONFIG_SENSOR_LOG_LEVEL_DBG=y

nashif commented on samples/sensor/mma8451q/prj.conf:
```diff
@@ -1,6 +1,6 @@
-CONFIG_SYS_LOG_SENSOR_LEVEL=1
-CONFIG_SYS_LOG=y
+CONFIG_LOG=y
+CONFIG_SENSOR_LOG_LEVEL=1
```
CONFIG_SENSOR_LOG_LEVEL_INF=y

nashif commented on samples/sensor/vl53l0x/prj.conf:
```diff
@@ -7,5 +7,5 @@ CONFIG_VL53L0X_PROXIMITY_THRESHOLD=100
-#CONFIG_SYS_LOG=y
-#CONFIG_SYS_LOG_SENSOR_LEVEL=4
+#CONFIG_LOG=y
+#CONFIG_SENSOR_LOG_LEVEL=4
```
CONFIG_SENSOR_LOG_LEVEL_DBG=y

nashif commented on tests/drivers/build_all/sensors_a_h.conf:
```diff
@@ -3,8 +3,8 @@ CONFIG_GPIO=y
-CONFIG_SYS_LOG=y
-CONFIG_SYS_LOG_SENSOR_LEVEL=4
+CONFIG_LOG=y
+CONFIG_SENSOR_LOG_LEVEL=4
```
CONFIG_SENSOR_LOG_LEVEL_DBG=y

punitvara commented:
@vanwinkeljan Done. @nashif  Reviews comments addressed. Can I move Incline function to .c file as this CI is failing or should I wait for #10215  to get merged ?

nashif committed:
sensors: lis2dh: move header after log module init

Allow logging in header file using the module of the file including the
header.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
